picketlink/src/main/java/org/wildfly/extension/picketlink/common/model/AbstractResourceDefinition.java

```diff
@@ -119,7 +119,7 @@ private void addChildResourceDefinition(ModelElement resourceDefinitionKey, Reso
             childResourceDefinitions.put(resourceDefinitionKey, childResources);
         }
 
-        if (!childResources.contains(resourceDefinitionKey)) {
+        if (!childResources.contains(resourceDefinition)) {
```

Comment:
Since childResources is an ArrayList, this "if" statement is completely redundant with the logic following it.

Reply:
Sorry - I should have been more clear...
By redundant, I meant at the if statement should go away.
Because the ResourceDefinitions in question do not implement equals(...), the contains(...) method will use reference equality to determine if the resource definition is contained in the list.
Since the logic below is more permissive, we want to remove the if statement and instead use the logic that follows it (which determines equality based on the ResourceDefinition's path key.

Reply:
ok, got that. NP to fix. Can you please remove the PR from the CI queue and ping me again?

Reply:
> can you remove the PR from the CI queue

Done.